test/e2e/dynamic_resource_e2e_test.go

```diff
@@ -3,103 +3,164 @@ package e2e
-
-		// Wait for the expected InstallPlan's execution to either fail or succeed
-		ipName := sub.Status.InstallPlanRef.Name
-		ip, err := waitForInstallPlan(GinkgoT(), crc, ipName, sub.GetNamespace(), buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseFailed, v1alpha1.InstallPlanPhaseComplete))
-		require.NoError(GinkgoT(), err)
-		// Ensure the InstallPlan contains the steps resolved from the bundle image
-		expectedSteps := map[registry.ResourceKey]struct{}{
-			{Name: "my-prometheusrule", Kind: "PrometheusRule"}: {},
-			{Name: "my-servicemonitor", Kind: "ServiceMonitor"}: {},
-		}
-
-		for _, step := range ip.Status.Plan {
-			key := registry.ResourceKey{
-				Name: step.Resource.Name,
-				Kind: step.Resource.Kind,
-			}
-			for expected := range expectedSteps {
-				if strings.HasPrefix(key.Name, expected.Name) && key.Kind == expected.Kind {
-					delete(expectedSteps, expected)
+
+		AfterEach(func() {
+
+			// Delete Namespace
+			err := c.KubernetesInterface().CoreV1().Namespaces().Delete(ns.GetName(), deleteOpts)
+			Expect(err).NotTo(HaveOccurred())
+
+		})
+
+		Context("when a CatalogSource is created with a bundle that contains prometheus objects", func() {
+			var (
+				catsrc *v1alpha1.CatalogSource
+			)
+			BeforeEach(func() {
+
+				// Create CatalogSource
+				catsrc = &v1alpha1.CatalogSource{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      genName("dynamic-catalog-"),
+						Namespace: ns.GetName(),
+					},
+					Spec: v1alpha1.CatalogSourceSpec{
+						Image:      "quay.io/olmtest/catsrc_dynamic_resources:e2e-test",
+						SourceType: v1alpha1.SourceTypeGrpc,
+					},
 				}
-			}
-		}
-		require.Lenf(GinkgoT(), expectedSteps, 0, "Resource steps do not match expected: %#v", expectedSteps)
-
-		// Confirm that the expected types exist
-		gvr := schema.GroupVersionResource{
-			Group:    "monitoring.coreos.com",
-			Version:  "v1",
-			Resource: "prometheusrules",
-		}
-
-		err = waitForGVR(dynamicClient, gvr, "my-prometheusrule", ns.GetName())
-		require.NoError(GinkgoT(), err)
-
-		gvr.Resource = "servicemonitors"
-		err = waitForGVR(dynamicClient, gvr, "my-servicemonitor", ns.GetName())
-		require.NoError(GinkgoT(), err)
+
+				catsrc, err = crc.OperatorsV1alpha1().CatalogSources(catsrc.GetNamespace()).Create(catsrc)
+				Expect(err).NotTo(HaveOccurred())
+
+				// Wait for the CatalogSource to be ready
+				catsrc, err = fetchCatalogOnStatus(crc, catsrc.GetName(), catsrc.GetNamespace(), catalogSourceRegistryPodSynced)
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+
+			AfterEach(func() {
+				// Delete CatalogSource
+				err = crc.OperatorsV1alpha1().CatalogSources(catsrc.GetNamespace()).Delete(catsrc.GetName(), deleteOpts)
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+
+			Context("should install the operator successfully", func() {
+				var (
+					subName    string
+					sub        *v1alpha1.Subscription
+					cleanupSub cleanupFunc
+				)
+
+				AfterEach(func() {
+					cleanupSub()
```

Comment:
This is fragile because it will panic if `cleanupSub` is not assigned a non-nil value during every spec that is added to this context. Maybe all three of the context-scoped variables belong inside the spec, since there is no common setup happening here.